src/CodeStyle/CSharp/CodeFixes/CSharpFormattingAnalyzerImpl.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.CodingConventions;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    internal class CSharpFormattingAnalyzerImpl : AbstractFormattingAnalyzerImpl
```

Comment:
what is the difference between a FormattingAnalyzer and a FormattingAnalyzerImpl?

Reply:
FormattingAnalyzerImpl depends on the Workspaces layer, but it's not available by default during compilation. FormattingAnalyzer offers an entry point that adds the assembly resolve handler prior to creating the instance of FormattingAnalyzerImpl.